fix: case-insensitive 'Content-type' response header detection

vch-7 opened the pull request:
Changed the `Content-Type` response header detection to be case-insensitive in the [getResult ](https://github.com/prisma-labs/graphql-request/blob/f5f7286748943665f969082dbe91aed3195e8b7a/src/index.ts#L537)function.

Fixes: #314

jasonkuhrt reviewed (approved):
Thanks, test would be great.

vch-7 commented:
Thanks!

To add more context, the primary use case for this would be when the default `fetch` config in the `GraphQLClient` instance is overridden with say, an Axios wrapper, where the Response Header object does not implement a case-insensitive `get` method by default. I've added this to the issue description as well

Added a test for this use case

jasonkuhrt commented:
@vch-7 Not sure yet, still need to look into failing release tool stuff.
